lib/libzrepl/mgmt_conn.c

```diff
@@ -598,6 +598,64 @@ finish_async_tasks(void)
 	return (rc);
 }
 
+/*
+ * Take a snapshot and update snapshot IO to ZAP.
+ */
+int
+uzfs_zvol_create_snapshot_update_zap(zvol_info_t *zinfo,
+    char *snap, uint64_t snapshot_io)
+{
+	int ret = 0;
+
+	mutex_enter(&zvol_list_mutex);
```

Comment:
We may need zinfo's mtx rather than global lock zvol_list_mutex

Reply:
Since timer thread update zap attribute under same lock, I deliberately used this lock. Otherwise it may race with timer thread and timer thread may update IO number which might be smaller than snapshot IO number. Not only that it may cause seriuos problem if both threads race when we are updating "zinfo->checkpointed_ionum" & "zinfo->checkpointed_ionum"  counters.

Reply:
As per our discussion, I am retaining this lock as it is.